roofit/roofitcore/test/testRooDataSet.cxx

```diff
@@ -44,3 +45,87 @@ TEST(RooDataSet, ImportFromTreeWithCut)
   ASSERT_NE(theX, nullptr);
   EXPECT_FLOAT_EQ(theX->getVal(), 1.337);
 }
+
+
+/// ROOT-9528 Branch names are capped after a certain number of characters
+TEST(RooDataSet, ImportLongBranchNames) {
+
+  TTree tree("theTree", "theTree");
+  double doub = 0.;
+  tree.Branch("HLT_mu6_mu4_bBmumux_BsmumuPhi_delayed_L1BPH_2M8_MU6MU4_BPH_0DR15_MU6MU4", &doub);
+  doub = 2.;
+  tree.Fill();
+  doub = 4.;
+  tree.Fill();
+
+  RooRealVar *v = new RooRealVar("HLT_mu6_mu4_bBmumux_BsmumuPhi_delayed_L1BPH_2M8_MU6MU4_BPH_0DR15_MU6MU4", "HLT_mu6_mu4_bBmumux_BsmumuPhi_delayed_L1BPH_2M8_MU6MU4_BPH_0DR15_MU6MU4", 0., -100000., 100000.);
+
+  RooDataSet ds("ds", "ds", RooArgSet(*v), RooFit::Import(tree));
+  EXPECT_EQ(static_cast<RooRealVar*>(ds.get(0)->find(*v))->getVal(), 2.);
+  EXPECT_EQ(static_cast<RooRealVar*>(ds.get(1)->find(*v))->getVal(), 4.);
+  EXPECT_EQ(ds.numEntries(), 2);
+  EXPECT_DOUBLE_EQ(ds.sumEntries("HLT_mu6_mu4_bBmumux_BsmumuPhi_delayed_L1BPH_2M8_MU6MU4_BPH_0DR15_MU6MU4 > 3."), 1.);
+}
+
+
+
+#include <TRandom3.h>
+#include <TH1F.h>
+#include <TCut.h>
```

Comment:
includes in the middle of the file are against ROOT coding convention afaik

Reply:
Should read that maybe. ;-)